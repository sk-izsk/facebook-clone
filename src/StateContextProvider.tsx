import React, { createContext, useContext, useReducer } from 'react';

export interface StateContextProviderProps {
  initialState?: any;
  reducer?: any;
}

const StateContext = createContext(undefined);

const StateContextProvider: React.FC<StateContextProviderProps> = ({ initialState, reducer, children }) => {
  return <StateContext.Provider value={useReducer(reducer, initialState) as any}>{children}</StateContext.Provider>;
};

const useStateValue: any = () => useContext(StateContext);

export { StateContextProvider, StateContext, useStateValue };
