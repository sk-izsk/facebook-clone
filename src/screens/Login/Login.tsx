import { Box, Button, makeStyles, Typography } from '@material-ui/core';
import React from 'react';
import { SiFacebook } from 'react-icons/si';
import { auth, googleAuthProvider } from '../../api';
import { useStateValue } from '../../StateContextProvider';
import { CustomTheme, theme } from '../../theme/muiTheme';
import { actionTypes } from '../../utils';

export interface LoginProps {}

const useStyles = makeStyles((theme: CustomTheme) => ({
  login: {
    width: '100vw',
    height: '100vh',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column',
  },
  loginLogo: {
    flex: 3,
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'column',
    justifyContent: 'space-evenly',
  },
  fbText: {
    color: theme.palette.primary.main,
    fontWeight: 600,
  },
  fbLogo: {},
  btn: {
    height: 50,
    width: 200,
    backgroundColor: theme.palette.primary.main,
    color: theme.palette.primary.light,
    fontSize: 20,
    borderRadius: 50,
    '&:hover': {
      backgroundColor: theme.palette.primary.main,
    },
  },
  buttonContainer: {
    flex: 2,
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
  },
}));

const Login: React.FC<LoginProps> = () => {
  const classes = useStyles();
  const [state, dispatch] = useStateValue();
  const signIn = async () => {
    try {
      const result: firebase.auth.UserCredential = await auth.signInWithPopup(googleAuthProvider);
      dispatch({ type: actionTypes.SET_USER, user: result.user });
      console.log('this is result', result.user);
    } catch (err) {
      alert(err.message);
    }
  };
  return (
    <Box className={classes.login}>
      <Box className={classes.loginLogo}>
        <SiFacebook className={classes.fbLogo} size={theme.spacing(15)} color={theme.palette.primary.main} />
        <Typography className={classes.fbText} variant='h2'>
          Facebook
        </Typography>
      </Box>
      <Box className={classes.buttonContainer}>
        <Button className={classes.btn} type='submit' onClick={signIn}>
          Sign In
        </Button>
      </Box>
    </Box>
  );
};

export { Login };
