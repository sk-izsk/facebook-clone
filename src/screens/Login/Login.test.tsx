import { render } from '@testing-library/react';
import React from 'react';
import { StateContextProvider } from '../../StateContextProvider';
import { Login } from './Login';

const mockUser = {
  image: 'fake image url',
  message: 'fake message',
  profilePic: 'fake profile pic',
  timestamp: 'fake time stamp',
  username: 'fake username',
};

test('Header renders correctly', () => {
  const { asFragment } = render(
    <StateContextProvider initialState={{ user: mockUser }}>
      <Login />
    </StateContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
