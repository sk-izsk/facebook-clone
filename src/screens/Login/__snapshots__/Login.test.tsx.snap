// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-7 makeStyles-login-1"
  >
    <div
      class="MuiBox-root MuiBox-root-8 makeStyles-loginLogo-2"
    >
      <svg
        class="makeStyles-fbLogo-4"
        color="#1877f2"
        fill="currentColor"
        height="120"
        role="img"
        size="120"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(24, 119, 242);"
        viewBox="0 0 24 24"
        width="120"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title />
        <path
          d="M23.9981 11.9991C23.9981 5.37216 18.626 0 11.9991 0C5.37216 0 0 5.37216 0 11.9991C0 17.9882 4.38789 22.9522 10.1242 23.8524V15.4676H7.07758V11.9991H10.1242V9.35553C10.1242 6.34826 11.9156 4.68714 14.6564 4.68714C15.9692 4.68714 17.3424 4.92149 17.3424 4.92149V7.87439H15.8294C14.3388 7.87439 13.8739 8.79933 13.8739 9.74824V11.9991H17.2018L16.6698 15.4676H13.8739V23.8524C19.6103 22.9522 23.9981 17.9882 23.9981 11.9991Z"
        />
      </svg>
      <h2
        class="MuiTypography-root makeStyles-fbText-3 MuiTypography-h2"
      >
        Facebook
      </h2>
    </div>
    <div
      class="MuiBox-root MuiBox-root-9 makeStyles-buttonContainer-6"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-5"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Sign In
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
